client/src/main/scala/ch.epfl.scala.index.client/Client.scala

```diff
@@ -141,7 +141,7 @@ object Client {
   }
 
   private def jumpToSearchInput(event: KeyboardEvent): Unit = {
-    if (event.keyCode == 191) {
+    if (event.keyCode == KeyCode.S) {
```

Comment:
Hm, I realized one thing that I had noticed earlier but now it’s going to become important. Using a character key as a shortcut requires carefully checking several conditions.

* Modifier keys: `!(event.ctrlKey || event.altKey || event.shiftKey || event.metaKey)`. Otherwise it may inadvertently hijack `Ctrl+S` and/or the Hi**s**tory menu.
* That the currently focused element is not one that would want the normal character. I believe a reasonably general check for that is (in Javascript talk) `!(event.target instanceof HTMLInputElement || event.target instanceof HTMLTextAreaElement || event.target.isContentEditable)`, although in specific cases you might get away with just testing that none of the known input fields are in focus.

Reply:
You're correct. I'll give it another go tomorrow.